dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/AppsecurityTest.java

```diff
@@ -0,0 +1,294 @@
+
+        // Use HttpClient to execute the redirected url
+        url = new URI("https://" + server.getHostname() + ":" + server.getHttpDefaultSecurePort() + redirect[1]);
+        getMethod = new HttpGet(url);
+        response = client.execute(getMethod);
+        assertEquals("Expected " + expectedCode + " status code for login", expectedCode, response.getStatusLine().getStatusCode());
+
+        String actual = "";
+        switch (expectedCode) {
+            case HttpServletResponse.SC_OK:
+                // Check the content of the response returned
+                actual = EntityUtils.toString(response.getEntity(), "UTF-8");
+                assertTrue("The url " + testUrl + " did not return the expected content",
+                           actual.contains("GROUP: " + group + ", NAME: " + userid + ", ROLES: " + expectedContent));
+                break;
+
+            case HttpServletResponse.SC_FORBIDDEN:
+                actual = EntityUtils.toString(response.getEntity(), "UTF-8");
+                assertTrue("The url " + testUrl + " did not return the expected content \""
+                           + expectedContent + "\"" + "The actual content was:\n" + actual, actual.contains(expectedContent));
+                break;
+
+            default:
+                break;
+
+        }
+    }
+
+    private static void initializeSSLContext() throws Exception {
+        SSLContext sslContext = SSLContext.getInstance("SSL");
+        sslContext.init(
+                        null,
+                        new TrustManager[] {
+                                             new X509TrustManager() {
+                                                 @Override
+                                                 public void checkClientTrusted(X509Certificate[] arg0,
+                                                                                String arg1) {
+                                                 }
+
+                                                 @Override
+                                                 public void checkServerTrusted(X509Certificate[] arg0,
+                                                                                String arg1) {
+                                                 }
+
+                                                 @Override
+                                                 public X509Certificate[] getAcceptedIssuers() {
+                                                     return null;
+                                                 }
+                                             }
+                        },
+                        new SecureRandom());
+        SSLContext.setDefault(sslContext);
+        HttpsURLConnection.setDefaultSSLSocketFactory(
+                                                      sslContext.getSocketFactory());
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        server.stopServer();
+        server.restoreServerConfiguration();
```

Comment:
Put `restoreServerConfiguration` in a finally block in case there is a failure to stop the server.

Reply:
done, thanks!